docs-en/chapter_array_and_linkedlist/list.md

```diff
@@ -1,21 +1,21 @@
 # List
 
-A "list" is an abstract data structure concept, representing an ordered collection of elements. It supports operations like element access, modification, addition, deletion, and traversal, without requiring users to consider capacity limitations. Lists can be implemented based on linked lists or arrays.
+A "list" is an abstract data structure concept, representing an ordered collection of elements. It provides operations such as element access, modification, addition, deletion, and iteration, without requiring users to consider capacity limitations. Lists can be implemented using linked lists or arrays.
 
-- A linked list naturally functions as a list, supporting operations for adding, deleting, searching, and modifying elements, and can dynamically adjust its size.
-- Arrays also support these operations, but due to their fixed length, they can be considered as a list with a length limit.
+- A linked list inherently serves as a list, allowing for the addition, deletion, searching, and modification of elements. Additionally, it has the flexibility to adjust its length dynamically.
+- Arrays also provide these operations, but due to their immutable length property, they can be considered as a list with a length limit.
 
-When using arrays to implement lists, **the fixed length property reduces the practicality of the list**. This is because we often cannot determine in advance how much data needs to be stored, making it difficult to choose an appropriate list length. If the length is too small, it may not meet the requirements; if too large, it may waste memory space.
+When implementing lists using arrays, **the immutable length property reduces their practicality**. This is due to the often unpredictable nature of data storage needs, making it challenging to select an appropriate list length. If the length is too small, it may not meet the requirements; if too large, it may waste memory space.
```

Comment:
immutable -> fixed. 
if requirement is to prioritize on simplicity.

optional:
This is due to the often unpredictable nature of data storage needs, making it challenging to select an appropriate list length. ->"This is because predicting the amount of data to be stored in advance is often challenging, making it difficult to choose an appropriate list length"

Reply:
I recommend refining the translation of this passage. The current translation is a bit cumbersome. I suggest a more concise version for better readability:

1. "the immutable length property reduces their practicality" -> "the immutability of length reduces the practicality of the list"

2. "This is due to ..." - > "This is because it's often challenging to determine in advance how much data needs to be stored, making it difficult to choose an appropriate list length."

Feel free to consider this suggestion for improvement.